packages/itwinui-react/src/core/Table/columns/actionColumn.tsx

```diff
@@ -94,45 +96,45 @@ export const ActionColumn = <T extends Record<string, unknown>>({
               });
             };
             return (
-              <MenuItem
+              <Checkbox
                 key={column.id}
-                startIcon={
-                  <Checkbox
-                    checked={checked}
-                    disabled={column.disableToggleVisibility}
-                    onClick={(e) => e.stopPropagation()}
-                    onChange={onClick}
-                    aria-labelledby={`iui-column-${column.id}`}
-                  />
-                }
-                onClick={onClick}
+                checked={checked}
                 disabled={column.disableToggleVisibility}
-              >
-                <div id={`iui-column-${column.id}`}>
-                  {column.render('Header')}
-                </div>
-              </MenuItem>
+                onChange={onChange}
+                label={column.render('Header')}
+              />
             );
           });
 
-      const dropdownMenuProps =
+      const popoverProps =
         typeof columnManager !== 'boolean'
           ? columnManager.dropdownMenuProps
           : {};
 
       return (
-        <DropdownMenu
-          {...dropdownMenuProps}
-          menuItems={headerCheckBoxes}
-          onVisibleChange={(open) => {
-            setIsOpen(open);
-            dropdownMenuProps?.onVisibleChange?.(open);
-          }}
+        <Popover
+          applyBackground
+          content={
+            <Flex
+              as='fieldset'
+              className='iui-fieldset-base iui-table-column-manager'
```

Comment:
Lets use this:

```suggestion
            <FieldsetBase
              as={Flex}
              className='iui-table-column-manager'
```

or


```suggestion
            <Flex
              as={FieldsetBase}
              className='iui-table-column-manager'
```

The `iui-fieldset-base` class is an implementation detail of `FieldsetBase`. Of course right now `FieldsetBase` just sets the class, but in the future we might add more things to it (similar to `ButtonBase`) so it's good to use the component.

Reply:
Actually this seems to not be working

![](https://github.com/iTwin/iTwinUI/assets/9084735/d2c3e52e-be56-40fc-afa9-96294ccfe021)

Reply:
Side note: I believe this is an easy mistake to make and in the future we should do something to make `<SomeComponent as={Flex}>` not override the rendered element.